openedx/core/djangoapps/oauth_dispatch/views.py

```diff
@@ -39,7 +40,7 @@ def get_adapter(self, request):
         client_id = self._get_client_id(request)
         monitoring_utils.set_custom_metric('oauth_client_id', client_id)
 
-        if dot_models.Application.objects.filter(client_id=client_id).exists():
+        if dot_models.Application.objects.filter(client_id=client_id).exists() or DISABLE_DOP_ADAPTER.is_enabled():
             monitoring_utils.set_custom_metric('oauth_adapter', 'dot')
             return self.dot_adapter
         else:
```

Comment:
1. I guess this doesn't need to be an `else`, because we are returning above.
2. In this final part of the function, I think we would check `DISABLE_DOP_ADAPTER` and either raise an exception or continue on to `return self.dop_adapter`.  But, maybe you can explain more, because maybe the dot_adapter would raise an appropriate error?